railib/api.py

```diff
@@ -356,8 +366,8 @@ def poll_with_specified_overhead(
             time.sleep(duration)
 
 
-def is_engine_term_state(state: str) -> bool:
-    return state == "PROVISIONED" or ("FAILED" in state)
+def is_engine_term_state(state: str, targetState: str) -> bool:
```

Comment:
I don't quite understand the change being made here, or at least I think this function name should change to be more clear what the intent is as it no longer just returns boolean 'if engine is in a term state'.  Since "PROVISIONED" is passed in lower down, whats the point here?

Reply:
oh I caught it... DELETED is also here.. so is DELETED not a term state? why wouldn't DELETED be added here?